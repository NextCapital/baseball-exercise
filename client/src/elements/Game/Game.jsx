import React from 'react';
import PropTypes from 'prop-types';

const Game = ({ awayTeam, homeTeam }) => (<div>{ `${awayTeam.name} @ ${homeTeam.name}` }</div>);

Game.propTypes = {
  awayTeam: PropTypes.object,
  homeTeam: PropTypes.object
}

export default Game;
