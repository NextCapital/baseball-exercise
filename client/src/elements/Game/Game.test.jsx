import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Game from './Game';

const renderer = new ShallowRenderer();

describe('Game', () => {
  let properties;

  beforeEach(() => {
    properties = {
      awayTeam: {
        name: 'Away Team'
      },
      homeTeam: {
        name: 'Home Team'
      }
    };
  });

  afterEach(() => {
    properties = null;
  });

  const render = () => {
    renderer.render(<Game {...properties }/>);
    return {
      component: renderer.getRenderOutput(),
      instance: renderer.getMountedInstance()
    };
  };

  describe('render', () => {
    test('renders a Game with the correct properties', () => {
      const { component } = render();
      expect(component).toMatchSnapshot();
    });
  });
});
