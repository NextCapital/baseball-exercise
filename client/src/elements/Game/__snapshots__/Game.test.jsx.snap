// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game render renders a Game with the correct properties 1`] = `
<div>
  Away Team @ Home Team
</div>
`;
